src/lando/main/tests/test_git.py

```diff
@@ -217,6 +218,53 @@ def test_GitSCM_apply_get_patch_merge(
     assert merge_patch_helper is None
 
 
+def test_GitSCM_apply_patch_bytes(git_repo: Path, git_patch: Callable):
+    scm = GitSCM(str(git_repo))
+
+    # Get patch content as bytes
+    patch_str = git_patch()
+    patch_bytes = patch_str.encode("utf-8")
+
+    # Apply patch using the new method
+    scm.apply_patch_bytes(patch_bytes)
+
+    commit = scm.describe_commit()
+
+    expected_patch = patch_str
+    new_patch = scm.get_patch(commit.hash)
+
+    assert new_patch, f"Empty patch unexpectedly generated for {commit.hash}"
+
+    # The git version stamp varies. Strip it from the output before comparing.
+    remove_git_version_re = r"\d+(\.\d+)+$"
+    no_version_patch = re.sub(remove_git_version_re, "", new_patch)
```

Comment:
I think we have to add a fixture for this next time round. I wrote the same logic in another test.

Reply:
nit: Given that this is repeated twice in this PR, I think now would be a good time to fix this (here and test_GitSCM_apply_patch_bytes_base64)